engines/dragons/dragons.cpp

```diff
@@ -1596,27 +1596,145 @@ void DragonsEngine::initSubtitleFlag() {
 	}
 }
 
-int centerText(const char *text) {
-	return 0x14 - (strlen(text) / 2 + 1);
+uint32 wideStrLen(uint16 *text) {
+	int i = 0;
+	while (text[i] != 0) {
+		i++;
+	}
+	return i;
+}
+
+int centerText(uint16 *text) {
+	return 0x14 - (wideStrLen(text) / 2 + 1);
 }
 
 void DragonsEngine::mainMenu() {
 	_inMenu = true;
-	//TODO need to support other languages.
-	const char copyright[6][40] = {
-			"Crystal Dynamics is a trademark",
-			"of Crystal Dynamics.",
-			"Blazing Dragons is a trademark and",
-			"copyright of Terry Jones and is",
-			"used with permission.",
-			"Licensed by Nelvana Marketing Inc."
+
+	int lang_index, y_offset;
+
+	int lastMenuItem = 1;
+
+	uint16 copyright[4][11][40] = {
```

Comment:
You don't need to convert these to uint16, English, German and French should work fine, and the non-English characters can be escaped.

Check how other engines are handling hardcoded strings, e.g. itedata.cpp in SAGA:

https://github.com/scummvm/scummvm/blob/master/engines/saga/itedata.cpp

Reply:
You basically write it like this:

"Blazing Dragons ist gesch\x81""tztes"

Reply:
Should be fine now :)